src/Tizen.NUI.Scene3D/src/public/ModelMotion/MotionIndex.cs

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright(c) 2022 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+using System;
+using System.ComponentModel;
+using System.Collections.Generic;
+using Tizen.NUI;
+
+namespace Tizen.NUI.Scene3D
+{
+    /// <summary>
+    /// Index of motion value. It will be used to specify the target of motion
+    ///
+    /// We can use this class below cases
+    /// - ModelNodeKey (string key) , PropertySlotId (string key) , SlotType (ViewProperty) : Target is ModelNode's ViewProperty
+    /// - ModelNodeKey (string key) , PropertySlotId (int key)    , SlotType (ViewProperty) : Target is ModelNode's ViewProperty [not implemented yet]
+    /// - ModelNodeKey (int key)    , PropertySlotId (string key) , SlotType (ViewProperty) : Target is ModelNode's ViewProperty [not implemented yet]
+    /// - ModelNodeKey (int key)    , PropertySlotId (int key)    , SlotType (ViewProperty) : Target is ModelNode's ViewProperty [not implemented yet]
+    /// - ModelNodeKey (string key) , PropertySlotId (string key) , SlotType (BlendShape)   : Target is ModelNode's BlendShape [not implemented yet]
+    /// - ModelNodeKey (string key) , PropertySlotId (int key)    , SlotType (BlendShape)   : Target is ModelNode's BlendShape
+    /// - ModelNodeKey (int key)    , PropertySlotId (string key) , SlotType (BlendShape)   : Target is ModelNode's BlendShape [not implemented yet]
+    /// - ModelNodeKey (int key)    , PropertySlotId (int key)    , SlotType (BlendShape)   : Target is ModelNode's BlendShape [not implemented yet]
+    /// - ModelNodeKey (null)       , PropertySlotId (string key) , SlotType (BlendShape)   : Target is all BlendShape [not implemented yet]
+    /// All other cases are invalid.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class MotionIndex : IDisposable
+    {
+        /// <summary>
+        /// TODO : Explaine me.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public enum PropertySlotType
```

Comment:
This patch is for Motion like character animation or BlendShape morphing.
But this looks to allow all of the ViewProperties those are animatable, right?
So, The role goes beyond motion, and it seems like another new NUI.Animation for View property.

And, to set SlotType field by user is slightly unfriendly api.
We may can use these way if you OK.

![MotionIndex drawio](https://user-images.githubusercontent.com/32317749/235875326-25497c40-39ea-4776-8061-0ff2e53e6eba.png)

User can simply select Motion type by creating proper child class.
And in the InternalGenerateMotionDataAnimation, we can simply check the class type.
And, GetPropertyName() can be used in Internal Model Class.

TransformType is just an example. If such a enum is used, the functionality is limited, but I think it will be a little easier to use.

Reply:
I think this architect is more useful then current `PropertySlotType`. Currently, user 'might' try to change only the `SlotType` value.  I'll apply this kind of class system.